components/s3_bucket/main.tf

```diff
@@ -62,6 +68,26 @@ resource "aws_s3_bucket" "bucket" {
     enabled = var.versioning || var.replication_target != null
   }
 
+  dynamic "lifecycle_rule" {
+    for_each = var.temporary_paths
+
+    content {
+      id      = md5(lifecycle_rule.value)
+      prefix  = lifecycle_rule.value
+      enabled = true
+
+      # Delete files in this path after two weeks:
+      expiration {
+        days = 14
+      }
+
+      # If versioned, delete non-current versions at this path after 3 months:
+      noncurrent_version_expiration {
+        days = 90
+      }
```

Comment:
Since we have versioning enabled on this production bucket, we'd end up with a lot of "deleted" version markers inside this `temporary/` directory. I figured it made sense to clean these up 90 days after the file itself is deleted. (We'll continue to retain other old versions indefinitely for safe-keeping).

Reply:
Ah missed this comment on my first pass!